extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/queries/InMemoryQueryRepository.java

```diff
@@ -173,7 +179,8 @@ public void delete(final UUID queryId) throws QueryRepositoryException {
     }
 
     @Override
-    public Set<StreamsQuery> list() throws QueryRepositoryException {
+    public Set<StreamsQuery> list() throws QueryRepositoryException, IllegalStateException {
+        checkState();
```

Comment:
Check state needs to happen inside of the lock everywhere because you're accessing the listeners list, which is lock protected.

Reply:
right.